### ISSUE TITLE
Can't get real-time collaboration to work

### ISSUE THREAD

**Issue by oxi** [Codiad/Codiad-Collaborative#10]
Hello,
since I have updated to the latest version, I can't get the collaboration-plugin to run. 
I can see that others are making changes in the file (the blue/red markers), but I can't see the code/text they are writing.
If such a mistake already known?
Greetings

**Comment by oxi** [comment]
https://github.com/Codiad/Codiad/issues/894

**Comment by paulocoghi** [comment]
I can confirm this issue with Codiad v2.7.2 and the latest version of Codiad-Collaborative (from 2016-03-18 12:58)

It is exactly the way @oxi mentioned, we can see the other's cursors, but we can't see the changes made.

**Comment by xilispe** [comment]
Getting the same sittuation here. This live code collaborative editing has ever worked?

**Comment by daeks** [comment]
Yeah it had worked and may still work. Original developers dropped it due to lag of time. Therefore volunteers are required to fix bugs, add features or rework the code

**Comment by Spiritus44** [comment]
Idem for me, codiad lost a big advantage...

**Comment by carpas** [comment]
Same here... this is the only thing preventing me from using Codiad. All  the rest is perfect!

**Comment by BurnedDesigns** [comment]
Has anyone been able to fix this issue?

**Comment by TheJamaican** [comment]
After trying to find the issue with this plugin, using Edge I've finally encountered an error in ace.js:
![screenshot 1](https://cloud.githubusercontent.com/assets/9295293/14762682/9bf76200-0978-11e6-8892-5e21d6c33f57.png)

The other log entries are from me trying to figure out why workers are not running in Mozilla Firefox Dev Edition 47.0a2

**Comment by TheJamaican** [comment]
After some poking around I've found the problem. The minified ace.js has different aliases for inserting and removing text from the editor, and looks for different indexes inside the object the server sends back with the changed file lines.
I'm not sure if I'm explaining it correctly but I'll see if I can fix this within the next hour or so. The minified ace.js file is extremely hard to understand.

**Comment by carpas** [comment]
We are cheering for you TheJamaican!! \o/

Best of luck o\

**Comment by Flolagale** [comment]
Hey @TheJamaican, since this plugin was originally written, the applyDeltas method of the Ace editor as changed. Now I think that you might have to use methods such as [insertInLine](https://github.com/ajaxorg/ace/blob/master/lib/ace/document.js#L285) which is available on the Ace document object. Good luck, you can fix it!

**Comment by Verdier** [comment]
Looks like applyDelta remains on the new API... probably easier to use it.

**Comment by halukyamaner** [comment]
Hello.

I am having the same issue, I can see the cursor of other users but not able to see changes real time.

Is there anyone solved this problem?

**Comment by daeks** [comment]
The current known workaround is:

Took an older ace.js file from Codiad 2.4.0
https://github.com/Codiad/Codiad/releases/tag/v.2.4.0

**Comment by Arctomium** [comment]
Did anyone find a fix for this? Would be much appreciated if I could get this feature.

**Comment by Arctomium** [comment]
@JamoDevNich I tried that but it doesn't work, it just says Warning: File is opened by username.

**Comment by Crytiqal** [comment]
Is this feature fixed?

**Comment by JamoDevNich** [comment]
@Crytiqal Unfortunately not(even though it works fine with the workaround), I haven't had time to attempt anything either since I've been so hung up on a big project.
I think it would be better for this to be rewritten since I've discovered some bugs with it, for example if the file is altered outside of the collaborative system, it is overwritten with the previous version once a user opens the file inside the collaborative system.

**Comment by X-Raym** [comment]
Same problem here. I hope this feature can be fixed soon, real time collaboration is a top feature for web based IDE. 🐱 
I understand that it be difficult, I wish good luck to those wo we digg this problem. Many thanks 👍

**Comment by Ryah** [comment]
The workaround works but for me removes all syntax highlighting along with html autocomplete (where it automatically closes tags, it's like it doesn't even detect the code being in html). This is slightly annoying to me since I collab a lot but also don't like typing when I don't have to. Anyone find any other workaround or is @JamoDevNich's workaround still the only one?

**Comment by DovaMerith** [comment]
Issue persists in 2018
